include/tudocomp/coders/AdaptiveHuffmanCoder.hpp

```diff
@@ -0,0 +1,857 @@
+#pragma once
+
+#include <algorithm>
+#include <bitset>
+#include <numeric>
+#include <queue>
+#include <cstring>
+#include <utility>
+#include <cmath>
+#include <string>
+#include <limits>
+
+#include <tudocomp/Env.hpp>
+#include <tudocomp/Coder.hpp>
+#include <tudocomp/util.hpp>
+#include <tudocomp/Range.hpp>
+#include <tudocomp/def.hpp>
+
+
+namespace tdc {
+
+    namespace adaphuff {
+
+#define SYMBOL 256
+#define NUMBER 512
```

Comment:
`uliteral_t` might not always be defined as a byte, so it would be better to derive constants like this from the `uliteral_t` typedef.

Eg, by using `tdc::ULITERAL_MAX` as a base value.